services/admin/src/store.ts

```diff
@@ -489,4 +536,276 @@ export class Store {
+
+    const writeInAdjudications = this.getWriteInAdjudicationRecords({
+      electionId,
+    });
+
+    return rows.map((row): Admin.WriteInSummaryEntry => {
+      const adjudication = writeInAdjudications.find(
+        (a) =>
+          a.contestId === row.contestId &&
+          a.transcribedValue === row.transcribedValue
+      );
+
+      return {
+        contestId: row.contestId,
+        transcribedValue: row.transcribedValue ?? undefined,
+        writeInCount: row.writeInCount,
+        writeInAdjudication: adjudication,
+      };
+    });
+  }
+
+  /**
+   * Gets all write-in records, filtered by the given options.
+   */
+  getWriteInRecords({
+    electionId,
+    contestId,
+    status,
+    limit,
+  }: {
+    electionId: Id;
+    contestId?: ContestId;
+    status?: Admin.WriteInAdjudicationStatus;
+    limit?: number;
+  }): Admin.WriteInRecord[] {
+    this.assertElectionExists(electionId);
+
+    const whereParts: string[] = ['cvr_files.election_id = ?'];
+    const params: Bindable[] = [electionId];
+
+    if (contestId) {
+      whereParts.push('contest_id = ?');
+      params.push(contestId);
+    }
+
+    if (status === 'adjudicated' || status === 'transcribed') {
+      whereParts.push('write_ins.transcribed_value is not null');
+    } else if (status === 'pending') {
+      whereParts.push('write_ins.transcribed_value is null');
+    }
+
+    if (typeof limit === 'number') {
+      params.push(limit);
+    }
+
+    const writeInRows = this.client.all(
+      `
+        select
+          write_ins.id as id,
+          write_ins.cvr_id as castVoteRecordId,
+          write_ins.contest_id as contestId,
+          write_ins.option_id as optionId,
+          write_ins.transcribed_value as transcribedValue,
+          write_ins.transcribed_at as transcribedAt
+        from write_ins
+        inner join
+          cvr_file_entries on write_ins.cvr_id = cvr_file_entries.cvr_id
+        inner join
+          cvr_files on cvr_file_entries.cvr_file_id = cvr_files.id
+        where
+          ${whereParts.join(' and ')}
+        ${typeof limit === 'number' ? 'limit ?' : ''}
+      `,
+      ...params
+    ) as Array<{
+      id: Id;
+      castVoteRecordId: Id;
+      contestId: ContestId;
+      optionId: ContestOptionId;
+      transcribedValue: string | null;
+      transcribedAt: Iso8601Timestamp | null;
+    }>;
+
+    const adjudicationRows = this.getWriteInAdjudicationRecords({ electionId });
+
+    return writeInRows
+      .map((row) => {
+        const adjudication = adjudicationRows.find(
+          (a) =>
+            a.contestId === row.contestId &&
+            a.transcribedValue === row.transcribedValue
+        );
+
+        const writeInRecord: Admin.WriteInRecord = {
+          id: row.id,
+          status: adjudication
+            ? 'adjudicated'
+            : typeof row.transcribedValue === 'string'
+            ? 'transcribed'
+            : 'pending',
+          castVoteRecordId: row.castVoteRecordId,
```

Comment:
Do you think the UI should just grab the actual CVR to get the image data from the electron storage, or will we need another endpoint to get the CVR data for an ID (or just return that in this endpoint, but since you are getting all of the write in rows for a contest that seems like it would explode the response too much maybe idk)

Reply:
This API endpoint is mostly meant to be used with a `limit` query param, likely of 1. So it could include that data.

Reply:
I'm a little confused how the limit works actually. If there are 5 different CVRs with a write in for a given contest, you can filter down to those 5 with the contestId param, and then limit to 1 but don't we need a paging param to then be able to get the other write ins?

Reply:
You'd also use `status=pending`, so as long as you're transcribing then the head of that response will always be different.